pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1442,3 +1442,20 @@ def test_merge_series(on, left_on, right_on, left_index, right_index, nm):
         with pytest.raises(ValueError, match='a Series without a name'):
             result = pd.merge(a, b, on=on, left_on=left_on, right_on=right_on,
                               left_index=left_index, right_index=right_index)
+
+
+@pytest.mark.parametrize("col1, col2, suffixes, expected_cols", [
+    (0, 0, ("", "_dup"), ["0", "0_dup"]),
+    (0, 0, (None, "_dup"), ["0", "0_dup"]),
+    (0, 0, ("_x", "_y"), ["0_x", "0_y"]),
+    ("a", 0, (None, "_y"), ["a", 0])
+])
+def test_merge_suffix(col1, col2, suffixes, expected_cols):
+    a = pd.DataFrame({col1: [1, 2, 3]})
```

Comment:
Can you add the github issue number as a comment?

Reply:
added!